dev/open.ts

```diff
@@ -1,7 +1,99 @@
 export const completionSpec: Fig.Spec = {
   name: "open",
   description: "open files using default application",
+  options: [
+    {
+      name: "-a",
+      description: "Specify the application to use for opening the file",
+      args: {
+        name: "Application",
+        generators: [
+          {
+            script: "ls -1p /Applications",
+            postProcess: function (out) {
+              return out.split("\n").map((line) => ({
+                name: line,
+                icon: `fig:///Applications/${line}`,
+                priority: line.endsWith(".app/") && 76,
+              }));
+            },
+          },
+          {
+            script: "ls -1p ~/Applications",
+            postProcess: function (out) {
+              return out.split("\n").map((line) => ({
+                name: line,
+                icon: `fig://~/Applications/${line}`,
+                priority: line.endsWith(".app/") && 76,
+              }));
+            },
```

Comment:
Those generators are amazing! But there is a bit of duplication, let's create a top-level function to generate a generator:

```ts
const appsGenerator = (path: string): Fig.Generator => ({
  script: `ls -1p ${path}`,
  postProcess: (out) => {
    return out.split("\n").map((line) => ({
      name: line,
      icon: `fig://${path}/${line}`,
      priority: line.endsWith(".app/") && 76,
    }));
  },
});
```

And we can use it like this:
```suggestion
          appsGenerator("/Applications"),
          appsGenerator("~/Applications"),
```

Reply:
Haha, just was about to push this up 😅